datafusion/sql/src/expr/value.rs

```diff
@@ -138,9 +138,19 @@ impl<'a, S: ContextProvider> SqlToRel<'a, S> {
                 schema,
                 &mut PlannerContext::new(),
             )?;
+
             match value {
                 Expr::Literal(scalar) => {
-                    values.push(scalar);
+                    values.push(Expr::Literal(scalar));
+                }
+                Expr::ScalarFunction(ref scalar_function) => {
+                    if scalar_function.fun == BuiltinScalarFunction::MakeArray {
+                        values.push(Expr::ScalarFunction(scalar_function.clone()));
```

Comment:
May we consider `values.push(value)` here?

Reply:
sure! thanks !